carta/util.py

```diff
@@ -129,3 +134,214 @@ def split_action_path(path):
     """
     parts = path.split('.')
     return '.'.join(parts[:-1]), parts[-1]
+
+
+class PixelString:
+    """Parses pixel values."""
+
+    UNITS = {"px", "pix", "pixel", "pixels"}
+    UNIT_REGEX = rf"^(\d+(?:\.\d+)?)\s*(?:{'|'.join(UNITS)})$"
+
+    @classmethod
+    def valid(cls, value):
+        """Whether the input string is a numeric value followed by a pixel unit.
+
+        Permitted pixel unit strings are stored in :obj:`carta.util.PixelString.UNITS`. Whitespace is permitted after the number and before the unit.
+
+        Parameters
+        ----------
+        value : string
+            The input string.
+
+        Returns
+        -------
+        boolean
+            Whether the input string is a pixel value.
+        """
+        m = re.match(cls.UNIT_REGEX, value, re.IGNORECASE)
+        return m is not None
+
+    @classmethod
+    def normalized(cls, value):
+        """Parse a string containing a numeric value followed by a pixel unit, and return the numeric part (as a string).
+
+        Permitted pixel unit strings are stored in :obj:`carta.util.PixelString.UNITS`. Whitespace is permitted after the number and before the unit.
+
+        Parameters
+        ----------
+        value : string
+            The string representation of the pixel value.
+
+        Returns
+        -------
+        string
+            The numeric portion of the pixel value.
+
+        Raises
+        ------
+        ValueError
+            If the input string is not in a recognized format.
+        """
+        m = re.match(cls.UNIT_REGEX, value, re.IGNORECASE)
+        if m is None:
+            raise ValueError(f"{repr(value)} is not in a recognized pixel format.")
+        return m.group(1)
+
+
+class AngularSizeString:
+    """Parses angular sizes."""
+
+    NORMALIZED_UNIT = {
+        "arcmin": "'",
```

Comment:
we can consider to also support arcsecond and arcminute. Sometimes, we also use "asec" for arcsec and "amin" for arcmin.

Reply:
These are all very easy to add.

Reply:
I added these, together with "arcseconds" and "arcminutes" ("degrees" is also supported, but only these full words can be plural, not any of the abbreviations).